pkg/controller/broadcastjob/broadcastjob_controller.go

```diff
@@ -653,6 +653,13 @@ func (r *ReconcileBroadcastJob) createPods(nodeName, namespace string, template
 	if err != nil {
 		return err
 	}
+	accessor, err := meta.Accessor(object)
+	if err != nil {
+		klog.Errorf("parentObject does not have ObjectMeta, %v", err)
+		return nil
+	}
+	// Specify the pod.Name to prevent the creation of two Pods in a Node
+	pod.Name = fmt.Sprintf("%s-%s", accessor.GetName(), nodeName)
```

Comment:
To avoid naming conflict, we should use the full name of the broadcastjob CR instead of using name only.

Reply:
I don't think it's necessary. Each object in your cluster has a Name that is unique for that type of resource. BroadcastJob itself wants to run only once on each Node, so a pod's name consist of the name of broadcastJob Name and nodeName is enough to avoid naming conflict.

Reply:
Broadcastjob is namespace scoped. So you can have two broadcastjob CRs with the same name but in different namespace. They will conflict when counting how many Pods has been running in each node.

Reply:
BroadcastJob is namespace scoped. So if you create two BroadcastJob CRs with the same name but in different namespace, they will conflict by the Pod name when creating Pod in each node.

Reply:
Pod is also namespace scoped. So it's ok if podName is same in different namespace.

Reply:
You are right. I take it back